stl/inc/format

```diff
@@ -1758,7 +1817,7 @@ private:
     void _Flush() {
         auto _Size = this->_Size();
         this->_Clear();
-        _Output = _STD _Copy_unchecked(_Data, _Data + this->_Limit(_Size), _STD move(_Output));
+        _Iterator_appender<_OutputIt, _Ty>{_Output}._Append(_Data, _Data + this->_Limit(_Size));

```

Comment:
If this is the only callsite, why have a class template `_Iterator_appender` at all instead if just a function template:
```c++
template <class _It, class _Ty>
concept _Appendable_back_inserter = _Is_specialization_v<_It, back_insert_iterator> && _Appendable_container<typename _It::container_type, _Ty>;

template <class _It, class _Ty>
concept _Insertable_back_inserter = _Is_specialization_v<_It, back_insert_iterator> && _Insertable_container<typename _It::container_type, _Ty>;

template <class _OutIt, class _Ty>
void _Meow(_OutIt& _Out, _Ty* _First, _Ty* _Last) {
    if constexpr (_Appendable_back_inserter<_OutIt, _Ty>) {
        using _Container = typename _OutIt::container_type;
        auto& _Cont = *_Back_insert_iterator_container_access<_Container>{_Out}.container;
        _Cont.append(_First, _Last);
    } else if constexpr (_Insertable_back_inserter<_OutIt, _Ty>) {
        using _Container = typename _OutIt::container_type;
        auto& _Cont = *_Back_insert_iterator_container_access<_Container>{_Out}.container;
        _Cont.insert(_RANGES end(_Cont), _First, _Last);
    } else {
        _Out = _STD _Copy_unchecked(_First, _Last, _STD move(_Out));
    }
}
```

Reply:
I was thinking it could be useful for people to specialize `_Iterator_appender` for their exotic containers (instead of taking everything with `T::append`, since we don't know if it behaves like a standard container's `append`), but maybe not a scenario we want to support, probably more suitable for a C++ paper.

Reply:
(and I forgot concepts allows that kind of thing)

Reply:
The _next_ issue I'm going to be harping on is that we need to constrain this to standard library containers only since we can't duck type in the standard library. `back_insert_iterator` works with anything that has `push_back`, there's no guarantee that `append` doesn't launch the missiles, or even that testing for the presence of `append` doesn't cause a compile error.

Reply:
But then we lose the optimization and go through the slow `push_back`-per-character approach for custom containers. This is a scenario I kinda care about since I use a custom container which uses the stack for < 512 characters, as a way to make formatting of small strings not require the heap, and losing this optimization would make me very sad :(

So if we keep `_Iterator_appender` (maybe insert a comment saying this can be specialized by users as an extension?), specialize it to vector and basic_string only, then I could specialize it for my own container and keep the benefits of this optimization.